src/Exception/ExceptionFactory.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace bunq\Exception;
+
+/**
+ */
+class ExceptionFactory
+{
+    /**
+     * HTTP error response codes constants.
+     */
+    const HTTP_RESPONSE_CODE_BAD_REQUEST = 400;
+    const HTTP_RESPONSE_CODE_UNAUTHORIZED = 401;
+    const HTTP_RESPONSE_CODE_FORBIDDEN = 403;
+    const HTTP_RESPONSE_CODE_NOT_FOUND = 404;
+    const HTTP_RESPONSE_CODE_METHOD_NOT_ALLOWED = 405;
+    const HTTP_RESPONSE_CODE_TOO_MANY_REQUESTS = 429;
+    const HTTP_RESPONSE_CODE_INTERNAL_SERVER_ERROR = 500;
+
+    /**
+     * The first item index in an array.
+     */
+    const INDEX_FIRST = 0;
+
+    /**
+     * @param string $message
+     * @param array $args
+     *
+     * @return BadRequestException|ForbiddenException|MethodNotAllowedException|NotFoundException|PleaseContactBunqException|ToManyRequestsException|UnauthorizedException|UnknownApiErrorException
+     */
+    public static function createExceptionForResponse($message, $args = [])
```

Comment:
Same question: do we need args here?

Also please add strict param types and the response type :)